pajbot/bot.py

```diff
@@ -130,16 +120,44 @@ def __init__(self, config: cfg.Config, args: argparse.Namespace) -> None:
         self.app_token_manager = AppAccessTokenManager(self.twitch_id_api, RedisManager.get())
         self.twitch_helix_api: TwitchHelixAPI = TwitchHelixAPI(RedisManager.get(), self.app_token_manager)
 
-        self.bot_user_id = self.twitch_helix_api.get_user_id(self.nickname)
+        # bot
+        if "bot_id" in config["main"]:
+            self.bot_user_id = config["main"]["bot_id"]
+            self.nickname = self.twitch_helix_api.get_login(self.bot_user_id)
+        elif "nickname" in config["main"]:
+            self.nickname = config["main"].get("nickname", "pajbot")
+            self.bot_user_id = self.twitch_helix_api.get_user_id(self.nickname)
+
+        # streamer
+        if "streamer_id" in config["main"]:
+            self.streamer_user_id = config["main"]["streamer_id"]
+            self.streamer = self.broadcaster = self.twitch_helix_api.get_login(self.streamer_user_id)
+            self.channel = f"#{self.streamer}"
+        else:
+            self.streamer, self.channel = cfg.load_streamer_and_channel(config)
+            self.broadcaster = self.twitch_helix_api.get_user_basics_by_login(self.streamer)
+            self.streamer_user_id = self.broadcaster.id
+
+        # streamer display
+        if "streamer_name" in config["web"]:
+            self.streamer_display = config["web"]["streamer_name"]
+        elif "streamer_name" not in config["web"]:
+            self.streamer_display = self.twitch_helix_api.get_display_name(self.streamer_user_id)
+
+        # control hub
+        if "control_hub_id" in config["main"]:
+            self.control_hub: Optional[str] = self.twitch_helix_api.get_login(config["main"]["control_hub_id"])
+        else:
+            self.control_hub: Optional[str] = config["main"].get("control_hub", None)
```

Comment:
The control_hub variable can be defined before we start checking if it exists in config["main"]
`self.control_hub: Optional[str]`